crates/services/consensus_module/poa/src/service.rs

```diff
@@ -480,53 +569,55 @@ where
             }
         }
 
-        if let SyncState::Synced(block_header) = &*state.borrow_and_update() {
-            let (last_height, last_timestamp, last_block_created) =
-                Self::extract_block_info(block_header);
-            if last_height > self.last_height {
-                self.last_height = last_height;
-                self.last_timestamp = last_timestamp;
-                self.last_block_created = last_block_created;
-            }
+        if let SyncState::Synced(block_header) = &*sync_state.borrow_and_update() {
+            self.update_last_block_values(block_header);
         }
 
-        tokio::select! {
-            biased;
-            _ = watcher.while_started() => {
-                should_continue = false;
-            }
-            request = self.request_receiver.recv() => {
-                if let Some(request) = request {
-                    match request {
-                        Request::ManualBlocks((block, response)) => {
-                            let result = self.produce_manual_blocks(block).await;
-                            let _ = response.send(result);
+        let next_height = self.next_height();
+        let chain_id = self.chain_id;
+        let maybe_block = self.predefined_blocks.get_block(&next_height);
+        if let Some(block) = maybe_block {
+            self.produce_predefined_block(&block, &chain_id).await?;
+            should_continue = true;
```

Comment:
Could we please make an early return here? To avoid shifting a big part of the code into the `if` statement?

We can move the code below into the `else {` in a separate PR